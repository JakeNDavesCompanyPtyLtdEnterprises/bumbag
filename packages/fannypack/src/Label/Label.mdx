---
name: Label
route: /form/label
menu: Form
---

import { Playground } from 'docz';
import PropsTable from '../_docs/components/PropsTable';
import { Box } from '../primitives/index';
import Input from '../Input';
import {Label} from './Label';

# Label

## Import

`import { Label } from 'fannypack'`

## Basic Usage

`<Label>` renders a `<label>` by default with basic reset styling.

<Playground>
  <Label>Hello world</Label>
  This is me
</Playground>

They can also be linked with any type of field, like `<Input>`.

<Playground>
  <Label htmlFor="name">Your name</Label>
  <Input id="name" />
</Playground>

## Accessibility

If using in conjunction with other fields, it is highly recommend to add a `htmlFor` prop to `<Label>` which links to the field `id`.

<Playground>
  <Label htmlFor="name">Your name</Label>
  <Input id="name" />
</Playground>

## Props

<PropsTable component={Label} />

## Theming

### Schema

```jsx
{
  Label: {
    base: css | Object
  }
}
```
