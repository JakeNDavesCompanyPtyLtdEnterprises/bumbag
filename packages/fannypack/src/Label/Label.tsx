import * as React from 'react';
import * as PropTypes from 'prop-types';
import { LabelProps as ReakitLabelProps } from 'reakit/ts';

import _Label from './styled';

export type LocalLabelProps = {
  children: React.ReactNode;
  className?: string;
  htmlFor?: string;
};
export type LabelProps = LocalLabelProps & ReakitLabelProps;

export const Label: React.FunctionComponent<LocalLabelProps> = ({ children, ...props }) => (
  <_Label {...props}>{children}</_Label>
);

Label.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  htmlFor: PropTypes.string
};
Label.defaultProps = {
  htmlFor: undefined,
  className: undefined
};

// @ts-ignore
const C: React.FunctionComponent<LabelProps> = Label;
export default Label;
