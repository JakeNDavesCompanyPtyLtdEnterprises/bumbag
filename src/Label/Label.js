// @flow
import React, { type Node } from 'react';

import _Label from './styled';

type Props = {
  use?: any,
  children: Node,
  className?: string,
  htmlFor?: string
};

const Label = ({ children, className, ...props }: Props) => (
  <_Label className={className} fontWeight="600" marginBottom="xxxsmall" {...props}>
    {children}
  </_Label>
);

Label.defaultProps = {
  use: undefined,
  htmlFor: undefined,
  className: undefined
};

export default Label;
