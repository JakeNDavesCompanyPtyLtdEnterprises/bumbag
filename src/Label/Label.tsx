import * as React from 'react';
// @ts-ignore
import PropTypes from 'prop-types';
import { LabelProps as ReakitLabelProps } from 'reakit/ts';

import _Label from './styled';

export interface LocalLabelProps {
  children: React.ReactNode;
  className?: string;
  htmlFor?: string;
}
export type LabelProps = LocalLabelProps & ReakitLabelProps;

export const Label: React.FunctionComponent<LocalLabelProps> = ({ children, className, ...props }) => (
  <_Label className={className} fontWeight="600" marginBottom="xxxsmall" {...props}>
    {children}
  </_Label>
);

Label.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  htmlFor: PropTypes.string
};
Label.defaultProps = {
  htmlFor: undefined,
  className: undefined
};

// @ts-ignore
const C: React.FunctionComponent<LabelProps> = Label;
export default Label;
