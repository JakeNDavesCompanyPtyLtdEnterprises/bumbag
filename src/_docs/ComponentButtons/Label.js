import React from 'react';

const SvgComponent = props => (
  <svg width={76} height={12} fill="none" {...props}>
    <path d="M75.564 0H0v3.649h75.564V0z" fill="#3926A5" />
    <path d="M37.782 7.663H0v3.649h37.782v-3.65z" fill="#3926A5" fillOpacity={0.12} />
  </svg>
);

export default SvgComponent;
