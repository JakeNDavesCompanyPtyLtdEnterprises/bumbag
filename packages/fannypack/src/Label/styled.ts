import Label from 'reakit/Label';
import { theme } from 'styled-tools';

import styled, { space } from '../styled';
import { LabelProps } from './Label';

export default styled(Label)<LabelProps>`
  display: block;
  font-weight: ${theme('fannypack.fontWeights.semibold')};
  margin-bottom: ${space(1)}rem;
  line-height: 1;

  & {
    ${theme('fannypack.Label.base')};
  }
`;
