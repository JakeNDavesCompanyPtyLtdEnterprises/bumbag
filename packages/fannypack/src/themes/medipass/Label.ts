import { css } from '../../styled';

export default {
  base: css`
    font-size: 14px;
  `
};
