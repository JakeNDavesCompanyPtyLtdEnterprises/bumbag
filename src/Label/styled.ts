import { Label } from 'reakit';
import { theme } from 'styled-tools';

import styled from '../styled';
import { LabelProps } from './Label';

export default styled(Label)<LabelProps>`
  display: block;

  ${theme('fannypack.Label.base')};
`;
