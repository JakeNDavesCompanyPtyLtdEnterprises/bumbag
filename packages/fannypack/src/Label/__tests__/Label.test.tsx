import * as React from 'react';
import render from '../../_utils/tests/render';
import Label from '../Label';

it('renders correctly for a basic label', () => {
  const { container } = render(<Label>Hello world</Label>);
  expect(container.firstChild).toMatchSnapshot();
});
