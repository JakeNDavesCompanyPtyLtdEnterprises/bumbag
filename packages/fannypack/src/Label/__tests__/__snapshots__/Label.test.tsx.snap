// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for a basic label 1`] = `
.c2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c2:focus:not(:focus-visible) {
  outline: none;
}

.c1 {
  display: inline-block;
}

.c0 {
  display: block;
  font-weight: 600;
  margin-bottom: 0.25rem;
  line-height: 1;
}

<label
  class="c0 c1 c2"
>
  Hello world
</label>
`;
